src/python/pants/option/global_options.py

```diff
@@ -929,6 +922,29 @@ def register_options(cls, register):
             help="Exclude target roots that match these regexes.",
         )
 
+        register(
+            "--files-depend-on-target-siblings",
+            type=bool,
+            default=False,
+            advanced=True,
+            help=(
+                "If true, every file in a target will automatically depend on the other files in "
+                "that same target. If false, you will either need to explicitly specify any "
+                "dependencies on sibling files (e.g. `./sibling.txt` in the `dependencies` field), "
```

Comment:
I wouldnt use `.txt` as an example, that's a bit confusing. I think `.py` is fine.